### PR TITLE
#220 fix for *.isInstance methods

### PR DESCRIPTION
**Pull request by wookieb**

### CODE REVIEW

File: test/either-spec.js [hunk 1]
```diff
@@ -311,6 +311,24 @@ describe('An Either', function () {
...
+        it('will return false for other monads', function () {
+            expect(Either.isInstance(Monet.Maybe.Some({}))).toBeFalsy();
+            expect(Either.isInstance(Monet.Maybe.None())).toBeFalsy();
+            expect(Either.isInstance(Monet.List.fromArray([]))).toBeFalsy();
+        })
+        it('will return false on non-monads', function () {
+            expect(Either.isInstance({})).toBeFalsy();
+            expect(Either.isInstance(true)).toBeFalsy();
+            expect(Either.isInstance(false)).toBeFalsy();
+            expect(Either.isInstance('foo')).toBeFalsy();
```
Comment (tomecko) [inline test/either-spec.js]: how about `.toBeTruthy()` -> `.toBe(true)` and `.toBeFalsy()` -> `.toBe(false)`?

File: test/io-spec.js [hunk 2]
```diff
@@ -48,8 +48,28 @@ describe('An IO monad', function () {
...
+            expect(effect.ap).toBe(effect['fantasy-land/ap'])
+        })
+    })
+
+    describe('IO.isInstance', function () {
+        it('will return true only for IO instances', function () {
+            var instance = IO(function () {
+                return 'foo'
+            })
+            expect(IO.isInstance(instance)).toBeTruthy();
```
Comment (tomecko) [inline test/io-spec.js]: it does not to be enforced but it seems the codebase follows no-semicolon style

File: test/either-spec.js [hunk 3]
```diff
@@ -311,6 +311,24 @@ describe('An Either', function () {
         })
     })
 
+    describe('Either.isInstance', function () {
+        it('will return true only for Either instances', function () {
```
Comment (tomecko) [inline test/either-spec.js]: nitpicking but I believe the test description should not contain the word "only" here, this particular test does not test negative cases

File: test/either-spec.js (hunk 1 above)
Comment (ulfryk) [reply test/either-spec.js]: IMO being explicit here is a good idea - output should match exactly `true` or `false` 👍

File: src/monet.js [hunk 4]
```diff
@@ -67,7 +67,7 @@
 
     function isInstance(typeName) {
         return function (target) {
-            return target[TYPE_KEY] || target.constructor[TYPE_KEY] === LIB_NAME + '/' + typeName
+            return (target[TYPE_KEY] || target.constructor[TYPE_KEY]) === LIB_NAME + '/' + typeName
```
Comment (ulfryk) [inline src/monet.js]: <3

File: src/monet.js [hunk 5]
```diff
@@ -1003,7 +1003,7 @@
                 throw new Error('IO requires a function.')
             }
             this.effectFn = effectFn
-            setType(Validation, TYPES_NAMES.IO)
+            setType(this, TYPES_NAMES.IO)
```
Comment (ulfryk) [inline src/monet.js]: :+1:

**Comment by ulfryk** [comment]
To be sure that the bug was fixed (and does not reappear in future) we need exact testing on all boolean values returned from checking methods. So as soon as all `truthy`/`falsy` checks are switched to `true`/`false` we can merge this one and release patch :)

**Comment by wookieb** [comment]
@ulfryk No problem. I'll do it later today once I finish work :)

**Pull request merged**